src/app/shared/button/button.directive.ts

```diff
@@ -0,0 +1,14 @@
+import { Directive, HostBinding, Input } from '@angular/core';
+
+@Directive({
+  selector: '[appButton]'
+})
+export class ButtonDirective {
+  @Input() color = 'neon-blue';
+  @Input() mode: 'default' | 'reject' = 'default';
+
+  @HostBinding('class') get colorClass() {
+    return ['button', this.mode === 'reject' ? 'button--reject' : `button--${this.color}`];
+  }
+  constructor() {}
+}
```

Comment:
Is it really necessary to have a own directive for the class handling here?
Couldn't this be solved by SASS classes?

Reply:
Well, I think it´s a matter of taste 😃. I prefer using simple directives for that instead of applying classes everywhere on all (standard) buttons. IMHO it makes the html more readable.

Reply:
I guess you are right with it being a matter of taste. 
But we can keep it like that 👍